resources/js/components/ui/calendar/TinyCalendar.tsx

```diff
@@ -0,0 +1,216 @@
+  getHoverRange?: (hoveredDay: DateTime) => DaysRange | undefined;
+  /** The function called when month is changed in the view. Controller should change the month prop. */
+  setMonth: (month: DateTime) => void;
+  /** The function called when a day is clicked. It should typically set selection. */
+  onDayClick?: (day: DateTime) => void;
+}
+
+const DEFAULT_PROPS = {
+  minDaysOfPrevMonth: 2,
+  minDaysOfNextMonth: 3,
+  minWeeks: 6,
+  showWeekdayNames: false,
+};
+
+interface DayInfo {
+  day: DateTime;
+  isToday: boolean;
+  classes: string;
+}
+
+export const TinyCalendar: VoidComponent<Props> = (props) => {
+  const mProps = mergeProps(DEFAULT_PROPS, props);
+  const [lProps, divProps] = splitProps(mProps, [
+    "selection",
+    "month",
+    "minDaysOfPrevMonth",
+    "minDaysOfNextMonth",
+    "minWeeks",
+    "showWeekdayNames",
+    "holidays",
+    "getHoverRange",
+    "setMonth",
+    "onDayClick",
+  ]);
+
+  const t = useLangFunc();
+  const weekDaysCalculator = createMemo(() => new WeekDaysCalculator(props.locale));
+  const weekInfo = () => weekDaysCalculator().weekInfo;
+  const monthStart = createMemo(() => lProps.month.startOf("month"), undefined, {
+    equals: (prev, next) => prev.toMillis() === next.toMillis(),
+  });
+  const [hover, setHover] = createSignal<DateTime>();
+  const hoverRange = createMemo<DaysRange | undefined>(() => {
+    const h = hover();
+    return h && (lProps.getHoverRange ? lProps.getHoverRange(h) : [h, h]);
+  });
+
+  const selectionCenter = () =>
+    lProps.selection && DateTime.fromMillis((lProps.selection[0].toMillis() + lProps.selection[1].toMillis()) / 2);
+  const retButtonAction = createMemo(() => {
+    if (lProps.month.hasSame(currentDate(), "month")) {
+      const selCenter = selectionCenter();
+      if (!selCenter) {
+        return undefined;
+      }
+      return lProps.month.hasSame(selCenter, "month") ? undefined : "toSelection";
+    } else {
+      return "toCurrent";
+    }
+  });
+
+  /** List of days to show in the calendar. */
+  const days = createMemo(() => {
+    const holidaysSet = new Set(lProps.holidays?.map((d) => d.startOf("day").toMillis()));
+    let day = weekDaysCalculator().startOfWeek(monthStart().minus({days: lProps.minDaysOfPrevMonth}));
+    const res: DayInfo[] = [];
+    function add(day: DateTime) {
+      const isToday = day.hasSame(currentDate(), "day");
+      res.push({
+        day,
+        isToday,
+        classes: cx({
+          [s.today!]: isToday,
+          [s.weekend!]: weekInfo().weekend.includes(day.weekday),
+          [s.holiday!]: holidaysSet.has(day.toMillis()),
+          [s.otherMonth!]: day.month !== monthStart().month,
+        }),
+      });
+    }
+    const last = monthStart().endOf("month").plus({days: lProps.minDaysOfNextMonth});
+    while (day <= last) {
+      add(day);
+      day = day.plus({days: 1});
+    }
+    while (res.length < 7 * lProps.minWeeks || day.weekday !== weekInfo().firstDay) {
```

Comment:
Potrzebne to minWeeks? Wydaje się zbędne.

A jeśli ma gwarantować stałą wysokość, to niepewne pod kątem maxWeeks.